index/src/index.rs

```diff
@@ -70,27 +86,47 @@ impl Index {
         &self.index
     }
 
+    pub fn searcher(&self) -> Result<Searcher> {
+        self.reader
+            .reload()
+            .map_err(|e| ReloadError(e.to_string()))?;
+        Ok(self.reader.searcher())
+    }
+
+    pub fn query_parser(&self) -> QueryParser {
+        QueryParser::for_index(&self.index, self.schema.default_fields())
+    }
+
     pub fn schema(self) -> SearchSchema {
         self.schema
     }
 
-    pub fn get_page_body(&self, page: u32, path: &str) -> Result<String> {
-        let doc = PdfDocument::load(format!("{}/{}", self.path, path))
-            .map_err(|_e| PdfNotFoundError(format!("{}/{}", self.path, path)))?;
+    pub fn get_page_body(&self, page: u32, path: impl AsRef<Path>) -> Result<String> {
+        let doc = PdfDocument::load(self.documents_path.join(path.as_ref())).map_err(|_e| {
+            PdfNotFoundError(self.documents_path.join(path).to_string_lossy().to_string())
+        })?;
         let text = doc.extract_text(&[page]).unwrap();
         Ok(text)
     }
 
-    fn create_index(path: String, schema: Schema) -> Result<TantivyIndex> {
+    fn create_index(path: &PathBuf, schema: Schema) -> Result<TantivyIndex> {
         TantivyIndex::create_in_dir(path, schema).map_err(|e| CreationError(e.to_string()))
     }
 
-    fn open_index(path: String) -> Result<TantivyIndex> {
+    fn open_index(path: &PathBuf) -> Result<TantivyIndex> {
         TantivyIndex::open_in_dir(path).map_err(|e| OpenError(e.to_string()))
     }
 
+    fn build_reader(index: &TantivyIndex) -> Result<IndexReader> {
+        index
+            .reader_builder()
+            .reload_policy(ReloadPolicy::OnCommit)
+            .try_into()
+            .map_err(|e| CreationError(e.to_string()))
+    }
```

Comment:
theoretically `ReloadPolicy::OnCommit` should no longer be needed. 
Although it might be worth leaving it as is, since in the case it *does* work after all the searcher can be created fast, leading to a better search time, since the reader does not have to reload everything. Thus in the long run tests to make sure whether this reload policy is applied and when it isn't would be good. For the PR we should leave it as is.

Reply:
Will be tackled in https://github.com/georgbuechner/litt/issues/11, give some feedback there if you like